RubberduckTests/Inspections/EnumerationDeclaredWithinWorksheetInspectionTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Linq;
+using Rubberduck.CodeAnalysis.Inspections;
+using Rubberduck.CodeAnalysis.Inspections.Concrete;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor.SafeComWrappers;
+using NUnit.Framework;
+
+namespace RubberduckTests.Inspections
+{
+    class EnumerationDeclaredWithinWorksheetInspectionTests : InspectionTestsBase
+    {
+        [Test]
+        [Category("Inspections")]
+        public void EnumerationDeclaredWithinWorksheet_InspectionName()
+        {
+            var inspection = new EnumerationDeclaredWithinWorksheetInspection(null, null);
+
+            Assert.AreEqual(nameof(EnumerationDeclaredWithinWorksheetInspection), inspection.Name);
+        }
+
+        [Test]
+        [Category("Inspections")]
+        [Category("EnumerationDeclaredWithinWorksheet")]
+        public void Project_with_multiple_enumerations_flags_only_enum_declared_within_worksheets()
+        {
+            #region InputCode
+            const string worksheetCode = @"Option Explicit
+Public Enum WorksheetEnum
+    wsMember1 = 0
+    wsMember1 = 1
+End Enum
+";
+            const string standardModuleCode = @"Option Explicit
+Public Enum StdModEnum
+    stdMember1 = 0
+    stdMember1 = 2
+End Enum
+";
+            const string classModuleCode = @"Option Explicit
+Public Enum ClassModEnum
+    classMember1 = 0
+    classMember2 = 3
+End Enum
+";
+            const string userFormModuleCode = @"Option Explicit
+Public Enum FormEnum
+    formMember1 = 0
+    formMember2 = 4
+End Enum
+";
+            #endregion
+
+            var inspectionResults = InspectionResultsForModules(
+                ("Sheet", worksheetCode, ComponentType.Document),
+                ("StdModule", standardModuleCode, ComponentType.StandardModule),
+                ("ClsMod", classModuleCode, ComponentType.ClassModule),
+                ("UserFormMod", userFormModuleCode, ComponentType.UserForm));
+
+            int actual = inspectionResults.Count();
+
+            Assert.AreEqual(1, actual);
+        }
+
+        [Test]
+        [Category("Inspections")]
+        [Category("EnumerationDeclaredWithinWorksheet")]
+        [TestCase(ComponentType.ActiveXDesigner)]
+        [TestCase(ComponentType.ClassModule)]
+        [TestCase(ComponentType.ComComponent)]
+        [TestCase(ComponentType.DocObject)]
+        [TestCase(ComponentType.MDIForm)]
+        [TestCase(ComponentType.PropPage)]
+        [TestCase(ComponentType.RelatedDocument)]
+        [TestCase(ComponentType.ResFile)]
+        [TestCase(ComponentType.StandardModule)]
+        [TestCase(ComponentType.Undefined)]
+        [TestCase(ComponentType.UserControl)]
+        [TestCase(ComponentType.UserForm)]
+        [TestCase(ComponentType.VBForm)]
+        public void Enumerations_declared_within_non_worksheet_object_have_no_inpsection_result(ComponentType componentType)
+        {
```

Comment:
Since the `IsResultDeclaration` logic simply uses the `ComponentType` parameter as-received in a single predicate expression, the test cases above are really tests of the `MockParser/InspectionBase` code rather unique execution paths of the inspection.  IMO you could consider removing these test cases since there are a few (and I think, sufficient) `false` scenarios tested in the preceding test.